src/Init/Data/Int/DivMod/Lemmas.lean

```diff
@@ -2354,6 +2354,76 @@ theorem bmod_neg_bmod : bmod (-(bmod x n)) n = bmod (-x) n := by
   apply (bmod_add_cancel_right x).mp
   rw [Int.add_left_neg, ← add_bmod_bmod, Int.add_left_neg]
 
+/- ### ediv -/
+
+theorem ediv_lt_self_of_pos_of_one_lt (x y : Int) (hx' : 0 < x) (hy' : 1 < y) :
+    x / y < x := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
+  apply Nat.div_lt_self (by omega) (by omega)
+
+-- note that x < y → x / y = 0
+theorem nonneg_ediv_of_pos_of_nonneg (x y : Int) (hx' : 0 < x) (hy' : 0 ≤ y) :
+    0 ≤ x / y := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
+  exact Int.ofNat_zero_le (xn / yn)
+
+/-!
+  needs a stricter bound on x, in fact (-1)/(-2) = 1 and 1 < (-1).natAbs is not true
+-/
+theorem ediv_lt_natAbs_self_of_lt_neg_one_of_lt_neg_one (x y : Int) (hx' : x < -1) (hy' : y < -1) :
+    x / y < x.natAbs := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_negSucc_of_lt_zero (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_negSucc_of_lt_zero (a := y) (by omega)
+  simp only [hx, hy, Nat.succ_eq_add_one, Int.ofNat_eq_coe, Int.natCast_add, Int.cast_ofNat_Int]
+  norm_cast
+  rw [natAbs_negSucc, Nat.succ_eq_add_one, Nat.add_lt_add_iff_right,
+    Nat.div_lt_iff_lt_mul (x := xn) (k := yn + 1) (y := xn) (by omega),
+    show (xn < xn * (yn + 1)) = (1 * xn < (yn + 1) * xn) by rw [Nat.one_mul, Nat.mul_comm]]
+  apply Nat.mul_lt_mul_of_lt_of_le (a := 1) (b := xn) (c := yn + 1) (d := xn) (by omega) (by omega) (by omega)
+
+theorem nonneg_ediv_of_neg_of_neg (x y : Int) (hx' : x < 0) (hy' : y < 0) :
+    0 ≤ x / y := by
```

Comment:
This already exists in a more general variant as `ediv_nonneg_of_nonpos_of_nonpos`.

Reply:
removed `nonneg_ediv_of_neg_of_neg`